src/aind_metadata_service/tars/mapping.py

```diff
@@ -117,41 +125,49 @@ def _map_virus_aliases(aliases: list) -> tuple[str, str, str]:
                 material_id = name
             else:
                 full_genome_name = name
-        # TODO:  check molecular registry for genome name with plasmid name
-        return plasmid_name, material_id, full_genome_name
+        viral_prep_aliases = ViralPrepAliases.model_construct(
+            plasmid_name=plasmid_name,
+            material_id=material_id,
+            full_genome_name=full_genome_name,
+        )
+        return viral_prep_aliases
 
-    def map_response_to_injection_materials(
-        self, response
-    ) -> Optional[List[InjectionMaterial]]:
+    @staticmethod
+    def map_full_genome_name(response, plasmid_name) -> str:
+        """Maps genome name from molecular response"""
+        full_genome_name = None
+        data = response.json()["data"][0]
+        aliases = data["aliases"]
+        for alias in aliases:
+            if alias["name"] != plasmid_name:
+                full_genome_name = alias["name"]
+        return full_genome_name
```

Comment:
Should the return type be Optional[str]? Also, what happens in the for loop? Can there be a break somewhere in there?

Reply:
The aliases in the data are usually length 2 (the 1st one usually the plasmid name, and the 2nd genome name but i also didn't want to assume that order). however, i didn't want to assume that the list is always length 2 hence the for loop for now. however (with assumption), it can be simplified like this:

 `if aliases[0]["name"] != plasmid_name:
    full_genome_name = aliases[0]["name"]
else:
    full_genome_name = aliases[1]["name"]`